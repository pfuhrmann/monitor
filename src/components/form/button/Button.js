import React from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';
import './Button.scss';

const Button = ({ children, className, ...otherProps }) => (
  <button className={cx('Form-btn', className)} {...otherProps}>{children}</button>
);

Button.propTypes = {
  children: PropTypes.node,
  className: PropTypes.string,
};

Button.defaultProps = {
  children: '',
  className: '',
};

export default Button;
