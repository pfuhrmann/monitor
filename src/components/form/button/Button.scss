@import "../../../variables.scss";

.Form-btn {
  display: flex;
  align-items: center;
  justify-content: center;
  height: 2.75em;
  font-size: inherit;
  background-color: $color-royal-blue;
  color: $color-white;
  border-width: 0;
  border-radius: $border-radius;
  width: 100% !important;
  vertical-align: middle;
  font-family: inherit;
  outline: none;
  padding: 0 1em;

  &:hover {
    background-color: $color-mariner;
    cursor: pointer;
  }
}
