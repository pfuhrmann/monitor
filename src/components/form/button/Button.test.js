import React from 'react';
import ReactDOM from 'react-dom';
import Button from './Button';
import renderer from 'react-test-renderer';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Button htmlFor="field">Some label</Button>, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renders button correctly', () => {
  const component = renderer.create(
    <Button htmlFor="field">Some label</Button>,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
