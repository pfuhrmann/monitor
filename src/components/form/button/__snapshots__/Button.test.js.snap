// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders button correctly 1`] = `
<button
  className="Form-btn"
  htmlFor="field"
>
  Some label
</button>
`;
